From 905113d0d3bd35970a3bd908eb8a36bcfafc3d1f Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Sun, 15 Jun 2014 21:46:19 +0300
Subject: [PATCH] Add zero padding to filename suffix in splitByFile()
Organization: Private
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Patch thanks to Alexandr Agranovsky <alesso@alesso.ru>

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 splitpatch.rb |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/splitpatch.rb
+++ b/splitpatch.rb
@@ -43,8 +43,10 @@
         if File.exists?(filename)
             puts "File #{filename} already exists. Renaming patch."
             appendix = 0
-            while File.exists?("#{filename}.#{appendix}")
+            zero = appendix.to_s.rjust(3, '0')
+            while File.exists?("#{filename}.#{zero}")
                 appendix += 1
+                zero = appendix.to_s.rjust(3, '0')
             end
             filename << ".#{appendix}"
         end
